Irmin-watcher and Core conflict

I tried to use Core (janestreet) with Irmin today and got a linking error.

```
xyz/_opam/lib/core/libcore_stubs.a(unix_stubs.o): In function `unix_realpath':
xyz/_opam/.opam-switch/build/core.v0.11.3/_build/default/src/unix_stubs.c:957: multiple definition of `unix_realpath'
xyz/_opam/lib/irmin-watcher/libirmin_watcher_stubs.a(realpath.o):xyz/_opam/.opam-switch/build/irmin-watcher.0.3.0/_build/default/src/realpath.c:45: first defined here
collect2: error: ld returned 1 exit status
File "caml_startup", line 1:
Error: Error during linking
```

--- a/src/core.ml
+++ b/src/core.ml
@@ -16,7 +16,7 @@ let stoppable t =
   Lwt.async (fun () -> Lwt.pick ([s; t ()]));
   function () -> Lwt.wakeup u (); Lwt.return_unit
 
-external unix_realpath : string -> string = "unix_realpath"
+external unix_realpath : string -> string = "irmin_watcher_unix_realpath"
 
 let realpath dir =
   let (/) x y = match y with

--- a/src/realpath.c
+++ b/src/realpath.c
@@ -28,7 +28,7 @@
 #include <caml/unixsupport.h>
 
 #ifdef WIN32
-CAMLprim value unix_realpath(value path)
+CAMLprim value irmin_watcher_unix_realpath(value path)
 {
   TCHAR buffer[PATH_MAX]=TEXT("");
   DWORD error = 0;
@@ -41,7 +41,7 @@ CAMLprim value unix_realpath(value path)
   return caml_copy_string(buffer);
 }
 #else
-CAMLprim value unix_realpath(value path)
+CAMLprim value irmin_watcher_unix_realpath(value path)
 {
   char buffer[PATH_MAX];
   char *r;

